src/traces/sankey/calc.js

```diff
@@ -17,8 +17,10 @@ var isIndex = Lib.isIndex;
 var Colorscale = require('../../components/colorscale');
 
 function convertToD3Sankey(trace) {
-    var nodeSpec = trace.node;
-    var linkSpec = trace.link;
+    // var nodeSpec = trace.node;
+    // var linkSpec = trace.link;
+    var nodeSpec = Lib.extendDeep({}, trace.node);
+    var linkSpec = Lib.extendDeep({}, trace.link);
```

Comment:
Do we really need to extend them? This can be slow.

Reply:
No, we actually don't need to do that! 

:hocho: in 1cd206f7fdba75b010650331fc1f7a726ec488df